cli/src/config.rs

```diff
@@ -135,19 +135,19 @@ impl LayeredConfigs {
     }
 
     #[instrument]
-    pub fn read_user_config(&mut self) -> Result<(), ConfigError> {
+    pub fn read_user_config(&mut self) -> Result<(), ConfigEnvError> {
```

Comment:
Optional: Could make all these return `Self` instead. Probably a good idea in general.

Reply:
`LayeredConfigs` isn't used in a builder-like way, so `Self -> Self` method is inconvenient. `&mut Self -> &mut Self` works, but the caller would have to allocate `Self` somewhere to get back owned object.